### DISCUSSION TITLE
Mes adaptations

### DISCUSSION THREAD

**Discussion by apprenti-sorcier** [augustinsoulard/KIT_BOTA_QFIELD#3] [Show and tell]
Voici comment j’adapte le projet pour l’utiliser selon mes besoins – si ça peut inspirer quelqu’un. Je les note comme dans un Wiki ; je compléterai quand je redécouvre autre chose.

- Vue / Décorations / Barre d’échelle en bas à gauche
- Dans Projet / Propriétés / Général, je renomme le projet, je m’assure que les unités de mesure de distance et de superficie sont en mètres et je passe le projet en Coordonnée SCR EPSG: 4326 - WGS 84 (je trouve que c’est la représentation graphique la plus familière et utilise des degrés décimaux plutôt que des mètres Lambert qui ne me disent rien)
- Dans la couche Flore, je redéfinis le symbole comme une étoile rouge de petite taille (1,6 mm) et les étiquettes de taille 5 ; ainsi on peut voir ça sur la carte, mais sans que ça cache le reste.
- Dans la couche Flore, je supprime le dernier champ auto‐incrémental « ID de la saisie » qui ne me sert à rien finalement. On peut retrouver l’ordre des saisies et donc du parcours par d’autres moyens. Par exemple, en définissant des étiquettes simples avec la structure suivante. « @id || '.' || "NomComplet" ». Une autre possibilité serait de convertir le champ « date » en « date et heure », mais ça demande des modifications plus en profondeur. Et une on peut même avoir une flèche du parcours avec une double symbologie
![Image du presse-papiers (104)](https://github.com/augustinsoulard/KIT_BOTA_QFIELD/assets/161721020/d39cebde-c747-46b4-995f-74af0d4cff0d)
(générateur de géométrie Polyligne  make_line(array_agg($geometry)) ; flèche à « remplissage simple / pas de remplissage », avec des lignes de 0,06 mm, largeurs de flèche de 0 pixels, longueur de pointe 9,5 mm, épaisseur de pointe 1,5 mm)
- La couche BDORTHO fausses couleurs en infrarouge ne me semble pas très utile… Je la désactive, mais j’ajoute les couches suivantes (dans l’ordre du haut vers le bas) :
  - des tableaux extraits des administrations en OpenData, comme par exemple les arbres plantés,
  - diverses superpositions comme la Base Adresses Nationale, BDTOPO (lieux‐dits et toponymie, surfaces hydrographiques, cours d’eau, les limites communales, de départements, de quartiers (https://geoservices.ign.fr/adminexpress), les mailles 5 × 5 km ou 10 × 10 km,
  - pour l’Île‐de‐France, la carte des végétations https://cbnbp.mnhn.fr/cbnbp/biodiversite/cartographieVegetationsIDF.jsp
  - beaucoup d’espaces naturels depuis la base INPN https://inpn-inspire.mnhn.fr/geoservices/MNHN/ows ou en téléchargement,
  - couche CORINE land cover https://www.data.gouv.fr/fr/datasets/corine-land-cover-occupation-des-sols-en-france/ (des styles sont fournis dans les bases régionales https://www.donnees.statistiques.developpement-durable.gouv.fr/donneesCLC/CLC/region/),
  - diverses images aériennes : BDOrtho historique, photog,aphies aériennes IGN historiques, Google satellite, ESRI, Bing satellite
  - cartes de l’état‐major du 19ᵉ siècle,
  - SCAN50 historique de 1950,
  - le plan IGN v2 (toutes sont dans le catalogue de données de l’IGN https://geoservices.ign.fr/catalogue) + Scan IGN
  - la carte OpenStreetMap de base, en activant le curseur d’opacité dans les propriétés / Légende + OpenTopoMap + couche Tracestack topo + OpenTopoMap
  - la Géologie / Pédologie, avec des couches des sols dominants dans la base de l’INRAE https://agroenvgeo.data.inra.fr/geoserver/web/?0 ou encore mieux les analyses régionales https://agroenvgeo.data.inra.fr/geonetwork/srv/fre/catalog.search 
  - les cartes géologiques 50k (dans la base du BRGM http://geoservices.brgm.fr/geologie),
  - ajout des altitudes, soit en France avec les courbes de niveau de l’IGN, soit à l’étranger par les fichiers SRTM HGT téléchargeables sur https://viewfinderpanoramas.org/dem3.html

Si on veut accélérer un peu les temps de transfert et de chargement des données, on peut ouvrir la base « DataProjetFlore.gpkg » dans un gestionnaire de bases de données qui peut supprimer les octets laissés à vide : on passe ainsi de 4,19 Mio à 1,49 Mio.
https://sqlitebrowser.org/ - Ensuite Outils / Compacter la base de données 

Quand on est à l’étranger, il faut penser à modifier au minimum la table Data / TaxaQGIS et les couches cartographiques avec celles du pays (à moins d’utiliser seulement OSM, qui a une carte mondiale).

**Comment by augustinsoulard** [comment]
Super ces propositions ! Je m'en inspirerais possiblement pour les futurs mises à jour tout en essayant de garder un projet adapté à un travail en bureau d'étude (car je sais que tu utilises moins les fonctionnalités Habitats, ZH et remarques). Tu as ajouté beaucoup de couches qui ont l'air top, il faut que je vois ce qui vaut le coup car je ne voudrais pas rendre le projet trop lourd. Il faut aussi que je pense à vider la BDD comme tu as fait.
Merci des astuces ;)

**Comment by apprenti-sorcier** [comment]
Précision : certaines couches (beaucoup d’espaces naturels de l’INPN, la cartographie CORINE land cover, les courbes de niveau du Géoportail, les fichiers d’altitude SRTM HGT, les végétations d’Île‐de‐France, les limites administratives, les données open data arbres…) peuvent être téléchargés pour une utilisation hors ligne.
Dans le dilemme projet léger avec des données à télécharger au besoin — utilisation complètement ou en grande partie hors ligne, économisant la batterie et la consommation de données au détriment de l’espace de stockage, à chacun de faire son choix.
